addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1256,7 +1256,13 @@ const Wysiwyg = Widget.extend({
     },
     _configureToolbar: function (options) {
         const $toolbar = this.toolbar.$el;
-        $toolbar.find('.btn-group').on('mousedown', e => e.preventDefault());
+        $toolbar.find('.btn-group').on('mousedown', e => {
+            // Do not prevent events on popovers.
+            if (e.target.closest('.dropdown-menu')) {
+                return;
+            }
+            e.preventDefault();
+        });
```

Comment:
I would have done:
```js
if (!e.target.closest('.dropdown-menu')) {
    e.preventDefault();
}
```
To be shorter and avoid to write an empty return. That being said, if you really do have an actual preference for the current version (I'd be interested to know why :p), I won't force you to change it.

Reply:
It felt like the check was a precondition for anything that could happen inside this handler - so my first intuition is to leave ASAP. Sure, inside this handler it currently is a single statement, but if it were to grow bigger, we'll gain one depth. (I did not actually take into account the fact that it uses one less "not".)
I can change it to your suggestion if you want, especially if it better matches the way things are usually done within `web_editor`.

Reply:
I would not like "exit ASAP" for the example you give here, where this condition is preceded by some code.
For the story, in a former job (2001~2010) in a very corporate environment, the internal rule was to check preconditions at the begin of functions, but with additional verbose boiler plate to have it extra-systematic.
```js
function f() {
    // Preconditions.
    if (condition) {
        return;
    }
    // Body.
    ... actual function code starts here...
    return result;
}
```
I believe all these precondition chunks might have been migrated outside the functions once aspect programming became available.
The body part on the other hand was strictly required to only contain one single return, at the end, of a local variable always named `result`.
It was a bit extreme...

So, I'll follow your advice and go for the short version here. :wink:

Reply:
Well I guess if you make it a de-facto standard in the entirety of the codebase then that weakens my argument because at that point you _know_ these preconditions exist and you know you have to check them. At that point it's a matter of personnal preference. Thanks for the change, I'm r+ing the PR.